src/main/scala/intellij/haskell/action/CreateHaskellFileAction.scala

```diff
@@ -50,6 +56,54 @@ class CreateHaskellFileAction extends CreateFileFromTemplateAction(CreateHaskell
     })
   }
 
+
+  override def createFileFromTemplate(originName: String, template: FileTemplate, originDir: PsiDirectory): PsiFile = {
+    val pathItems = HaskellFileUtil.getPathFromSourceRoot(originDir.getProject, originDir.getVirtualFile).orNull
```

Comment:
It's safer to use `originDir.getContainingFile` because that file is not virtual.